test/functional/eval/msgpack_functions_spec.lua

```diff
@@ -337,7 +337,8 @@ describe('msgpack*() functions', function()
     eq(1, eval('type(parsed[0]) == type(0) ' ..
                '|| parsed[0]._TYPE is v:msgpack_types.integer'))
     if eval('type(parsed[0]) == type(0)') == 1 then
-      eq(1, eval('0xFFFFFFFFFFFFFFFF == parsed[0]'))
+      command('call assert_equal(0xFFFFFFFFFFFFFFFF, parsed[0])')
```

Comment:
Wondering whether I needed casts in

```C
      if (mobj.via.u64 <= VARNUMBER_MAX) {
…
      if (mobj.via.i64 >= VARNUMBER_MIN) {
 ```

? AFAIR that should be upcasted automatically because `int64_t` rank is greater then `int` (equal with 64-bit `varnumber_T`), `uint64_t` is greater then `int` and `int64_t` should be promoted to `uint64_t` when comparing.